mmdet/models/mask_heads/fcn_mask_head.py

```diff
@@ -20,27 +21,30 @@ def __init__(self,
                  in_channels=256,
                  conv_kernel_size=3,
                  conv_out_channels=256,
-                 upsample_method='deconv',
-                 upsample_ratio=2,
                  num_classes=81,
                  class_agnostic=False,
+                 upsample_cfg=dict(type='deconv', upsample_ratio=2),
                  conv_cfg=None,
                  norm_cfg=None,
                  loss_mask=dict(
                      type='CrossEntropyLoss', use_mask=True, loss_weight=1.0)):
         super(FCNMaskHead, self).__init__()
-        if upsample_method not in [None, 'deconv', 'nearest', 'bilinear']:
+        self.upsample_cfg = upsample_cfg.copy()
+        if self.upsample_cfg['type'] not in [
+                None, 'deconv', 'nearest', 'bilinear', 'carafe'
+        ]:
             raise ValueError(
                 'Invalid upsample method {}, accepted methods '
-                'are "deconv", "nearest", "bilinear"'.format(upsample_method))
+                'are "deconv", "nearest", "bilinear", "carafe"'.format(
+                    self.upsample_cfg['type']))
         self.num_convs = num_convs
         # WARN: roi_feat_size is reserved and not used
         self.roi_feat_size = _pair(roi_feat_size)
         self.in_channels = in_channels
         self.conv_kernel_size = conv_kernel_size
         self.conv_out_channels = conv_out_channels
-        self.upsample_method = upsample_method
-        self.upsample_ratio = upsample_ratio
+        self.upsample_method = self.upsample_cfg.pop('type')
+        self.upsample_ratio = self.upsample_cfg.pop('upsample_ratio')
```

Comment:
We may consider renaming `upsample_ratio` to `scale_factor` since all upsample operators below have the argument `scale_factor`.

It will be simpler to manipulate the args.
```python
self.upsample_method = upsample_cfg_.get('type')

if self.upsample_method is None:
    self.upsample = None
elif self.upsample_method == 'deconv':
    upsample_cfg_.update(
        in_channels=upsample_in_channels,
        out_channels=self.conv_out_channels,
        kernel_size=self.upsample_ratio,
        stride=self.upsample_ratio)
elif self.upsample_method == 'carafe':
    upsampler_cfg_.update(
        channels=upsample_in_channels,
        scale_factor=self.upsample_ratio)
else:
    xxxxx
```